### ISSUE TITLE
Runs with error

### ISSUE THREAD

**Issue by tecosaur** [danylo-dubinin/zsh-jupyter-kernel#3]
Hello, I'd love to get this working

I installed with
```
python3 -mpip install notebook zsh_jupyter_kernel --user
python3 -mzsh_jupyter_kernel.install --prefix=/home/tec/.local
```
However now it doesn't work, and whenever I try it I get this:
```py
Traceback (most recent call last):
  File "/usr/lib64/python3.6/site-packages/pexpect/expect.py", line 111, in expect_loop
    incoming = spawn.read_nonblocking(spawn.maxread, timeout)
  File "/usr/lib64/python3.6/site-packages/pexpect/pty_spawn.py", line 509, in read_nonblocking
    raise TIMEOUT('Timeout exceeded.')
pexpect.exceptions.TIMEOUT: Timeout exceeded.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/tec/.local/lib/python3.6/site-packages/zsh_jupyter_kernel/__main__.py", line 4, in <module>
    IPKernelApp.launch_instance(kernel_class = ZshKernel)
  File "/usr/lib64/python3.6/site-packages/traitlets/config/application.py", line 657, in launch_instance
    app.initialize(argv)
  File "</usr/lib64/python3.6/site-packages/decorator.py:decorator-gen-124>", line 2, in initialize
  File "/usr/lib64/python3.6/site-packages/traitlets/config/application.py", line 87, in catch_config_error
    return method(app, *args, **kwargs)
  File "/usr/lib64/python3.6/site-packages/ipykernel/kernelapp.py", line 542, in initialize
    self.init_kernel()
  File "/usr/lib64/python3.6/site-packages/ipykernel/kernelapp.py", line 447, in init_kernel
    user_ns=self.user_ns,
  File "/usr/lib64/python3.6/site-packages/traitlets/config/configurable.py", line 412, in instance
    inst = cls(*args, **kwargs)
  File "/home/tec/.local/lib/python3.6/site-packages/zsh_jupyter_kernel/kernel.py", line 71, in __init__
    self._init_zsh_()
  File "/home/tec/.local/lib/python3.6/site-packages/zsh_jupyter_kernel/kernel.py", line 64, in _init_zsh_
    self.child.expect_exact(self.prompts['PS1'])
  File "/usr/lib64/python3.6/site-packages/pexpect/spawnbase.py", line 418, in expect_exact
    return exp.expect_loop(timeout)
  File "/usr/lib64/python3.6/site-packages/pexpect/expect.py", line 119, in expect_loop
    return self.timeout(e)
  File "/usr/lib64/python3.6/site-packages/pexpect/expect.py", line 82, in timeout
    raise TIMEOUT(msg)
pexpect.exceptions.TIMEOUT: Timeout exceeded.
<pexpect.pty_spawn.spawn object at 0x7f39fcf57908>
command: /bin/zsh
args: [b'/bin/zsh', b'-o', b'INTERACTIVE', b'-o', b'NO_ZLE', b'-o', b'NO_BEEP', b'-o', b'TRANSIENT_RPROMPT', b'-o', b'NO_PROMPT_CR', b'-o', b'INTERACTIVE_COMMENTS']
buffer (last 100 chars): ' \r\x1b[34m╭─\x1b[37m\ue0b2\x1b[39m\x1b[49m\x1b[47m \x1b[34m\uf17c \x1b[44m\x1b[37m\ue0b0 \x1b[30m\uf015 \x1b[30m \ue0b1 Desktop \x1b[49m\x1b[34m\ue0b0\x1b[39m \r\n\x1b[34m╰> '
before (last 100 chars): ' \r\x1b[34m╭─\x1b[37m\ue0b2\x1b[39m\x1b[49m\x1b[47m \x1b[34m\uf17c \x1b[44m\x1b[37m\ue0b0 \x1b[30m\uf015 \x1b[30m \ue0b1 Desktop \x1b[49m\x1b[34m\ue0b0\x1b[39m \r\n\x1b[34m╰> '
after: <class 'pexpect.exceptions.TIMEOUT'>
match: None
match_index: None
exitstatus: None
flag_eof: False
pid: 32586
child_fd: 52
closed: False
timeout: 5
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: <_io.TextIOWrapper name='/home/tec/.local/lib/python3.6/site-packages/zsh_jupyter_kernel/log/pexpect.log' mode='a' encoding='UTF-8'>
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0.05
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_string:
    0: 'PEXPECT_PS1 > '
[IPKernelApp] Cleaning up sockets
```

I have zsh 5.7.1, and python 3.6.9

Thanks in advance!

**Comment by danylo-dubinin** [comment]
Hi, @tecosaur. Which command are you trying to run?

**Comment by tecosaur** [comment]
Erm. `echo 1`

**Comment by danylo-dubinin** [comment]
Thanks. Okay I have found some info to work with. It happens in your case on initializing Zsh and it looks like something to do with the prompt, before any command can be executed.

The way I configure Zsh in the backend is to keep all your `.zshrc` configs but to reset prompts (`PS1`, `PS2`, ...) first to a predefined values and to stub your `precmd` function, which I know is usually used to set the goodies in prompts. It is required to be able to work with Pexpect, the tool which operates the backend Zsh.

Unfortunately, my init command was not successful in resetting your prompt values. After it run, it the prompt looked something like the following again:
```
╭─ ... Desktop ...
╰>
```

Do you use any other way to set the prompts? And/or let me know what kind of Zsh frameworks are you using like Oh My ZSH, etc.

**Comment by tecosaur** [comment]
Well in my case I'm using https://github.com/Powerlevel9k/powerlevel9k on top of _oh my zsh_ and the config is set with a bunch of `POWERLEVEL9K_XYZ` commands, for example in my case
```
ZSH_THEME="powerlevel9k/powerlevel9k"
POWERLEVEL9K_MODE="nerdfont-complete"

POWERLEVEL9K_PROMPT_ON_NEWLINE=true
# POWERLEVEL9K_RPROMPT_ON_NEWLINE=true

POWERLEVEL9K_MULTILINE_FIRST_PROMPT_PREFIX="%F{04}╭─%F{07}"
POWERLEVEL9K_MULTILINE_LAST_PROMPT_PREFIX="%F{04}╰> "

...

POWERLEVEL9K_LEFT_PROMPT_ELEMENTS=(os_icon root_indicator context dir dir_writable vcs)
POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS=(status background_jobs command_execution_time time)
```

**Comment by danylo-dubinin** [comment]
Thank you for details 👍 They were useful and I've fixed the issue in **version 3.1**.
Please update the package version.

Tech details: 
Indeed I have missed in Zsh docs one way in which frameworks set prompts: by adding hooks to precmd and preexec. #6

**Comment by tecosaur** [comment]
Now working! Thanks for fixing this so speedily.
